### PR TITLE
VM-868: Vendor details

### PR DESCRIPTION
**Pull request by asvishnyakov**
## Description
* **Breaking change:** `*MemberAddresses` command now return connection for `addresses` field instead of array
  * Don't used in vc-theme-b2b-vue, so just FYI
```graphql
mutation($command: InputUpdateMemberAddressType!) {
  updateMemberAddresses (command: $command) {
    addresses {
      items {        
        postalCode
        countryName
        regionName
        city
        line1
        line2
        name
      }
    }
  }
}
```
* Removed not used *.Core project
* Introduced https://virtocommerce.atlassian.net/browse/PT-10394
* Added extension for any type derived from `Member` registration in XAPI
* Added base query for queries to get specific types derived from `Member`
* **Check backward compatibility:** `contact`, `organization` queries should:
  * Be registered in schema
  * Have same arguments as earlier
  * Have same fields as earlier
  * Pass anonymous/registered user security limitations
  * Successfully return data as earlier
* Added new query `Vendor`
```graphql
query($vendorId: String!) {
  vendor (id: $vendorId) {
    id
    outerId
    memberType
    name
    status
    phones
    emails
    groups
    defaultBillingAddress {
      postalCode
      countryName
      regionName
      city
      line1
      line2
    }
    defaultShippingAddress {
      postalCode
      countryName
      regionName
      city
      line1
      line2
    }
    addresses {
      items {
        postalCode
        countryName
        regionName
        city
        line1
        line2
        name
      }
    }
    dynamicProperties {
      name
      valueType
      value
      dictionaryItem
      dynamicProperty
    }
    about
    iconUrl
    siteUrl
  }
}
```
### Jira-link:

https://virtocommerce.atlassian.net/browse/VM-868

### CODE REVIEW

File: src/VirtoCommerce.ProfileExperienceApiModule.Data/VirtoCommerce.ProfileExperienceApiModule.Data.csproj [hunk 1]
```diff
@@ -24,10 +24,14 @@
...
-    <PackageReference Include="VirtoCommerce.ExperienceApiModule.Core" Version="3.237.0" />
+    <PackageReference Include="VirtoCommerce.ExperienceApiModule.Core" Version="3.253.0-alpha.1090-vm-868" />
     <PackageReference Include="VirtoCommerce.MarketingModule.Core" Version="3.200.0" />
     <PackageReference Include="VirtoCommerce.Platform.Security" Version="3.259.0" />
     <PackageReference Include="VirtoCommerce.PricingModule.Core" Version="3.200.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Aggregates\NewFolder\" />
+  </ItemGroup>
```
Comment (ksavosteev) [inline src/VirtoCommerce.ProfileExperienceApiModule.Data/VirtoCommerce.ProfileExperienceApiModule.Data.csproj:35]: remove

File: src/VirtoCommerce.ProfileExperienceApiModule.Data/Middlewares/EmptyMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Threading.Tasks;
+using PipelineNet.Middleware;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Aggregates.Vendor;
+
+namespace VirtoCommerce.ProfileExperienceApiModule.Data.Middlewares;
+
+public class EmptyMiddleware: IAsyncMiddleware<VendorAggregate>
```
Comment (ksavosteev) [inline src/VirtoCommerce.ProfileExperienceApiModule.Data/Middlewares/EmptyMiddleware.cs:8]: add comments for why this is needed or remove it if not

File: src/VirtoCommerce.ProfileExperienceApiModule.Data/Extensions/SchemaExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+using VirtoCommerce.ExperienceApiModule.Core.Helpers;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Aggregates;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Queries;
+using VirtoCommerce.ProfileExperienceApiModule.Data.Schemas;
+
+namespace VirtoCommerce.ProfileExperienceApiModule.Data.Extensions;
+
+public static class SchemaExtensions
+{
+    public static void AddMemberQuery<TAggregate, TType, TQuery>(this ISchema schema, IMediator mediator, string name, Func<string, object, Task> checkAuthAsync)
```
Comment (ksavosteev) [inline src/VirtoCommerce.ProfileExperienceApiModule.Data/Extensions/SchemaExtensions.cs:17]: Use Query base class from Xapi Core if you want to refactor schema
https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/VirtoCommerce.ExperienceApiModule.Core/BaseQueries/Query.cs
Example: https://github.com/VirtoCommerce/vc-module-quote/tree/dev/src/VirtoCommerce.QuoteModule.ExperienceApi/Queries

**Review by ksavosteev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.ProfileExperienceApiModule.Data/VirtoCommerce.ProfileExperienceApiModule.Data.csproj (hunk 1 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.ProfileExperienceApiModule.Data/VirtoCommerce.ProfileExperienceApiModule.Data.csproj]: Fixed

File: src/VirtoCommerce.ProfileExperienceApiModule.Data/Middlewares/EmptyMiddleware.cs (hunk 2 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.ProfileExperienceApiModule.Data/Middlewares/EmptyMiddleware.cs]: Fixed

File: src/VirtoCommerce.ProfileExperienceApiModule.Data/Extensions/SchemaExtensions.cs (hunk 3 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.ProfileExperienceApiModule.Data/Extensions/SchemaExtensions.cs:17]: I know and like it, but refactored queries has `CheckAuthAsync` call, which should be refactored first. It's calling in ProfileSchema **22 times**, so it's not something I want to refactor now. I avoided code duplication for newly introduced Vendor queries and introduced base queries for members, they can be inherited from base quires of XAPI later.

**Review by ksavosteev — APPROVED** [review]

**Pull request merged**